components/electric/lib/electric/satellite/protobuf.ex

```diff
@@ -2,88 +2,101 @@ defmodule Electric.Satellite.Protobuf do
   # This is a version provided in the corresponding protocol buffer file
   # Make sure to bump it here and in the using macro below.
 
+  alias Electric.Satellite
+
   alias Electric.Satellite.{
     SatErrorResp,
-    SatAuthReq,
-    SatAuthResp,
-    SatInStartReplicationReq,
-    SatInStartReplicationResp,
-    SatInStopReplicationReq,
-    SatInStopReplicationResp,
     SatOpLog,
     SatRelation,
-    SatMigrationNotification,
-    SatSubsReq,
-    SatSubsResp,
     SatSubsDataError,
     SatSubsDataBegin,
     SatSubsDataEnd,
     SatShapeDataBegin,
     SatShapeDataEnd,
-    SatUnsubsReq,
-    SatUnsubsResp
+    SatRpcRequest,
+    SatRpcResponse
   }
 
   require Logger
 
-  @reserved [3, 4]
+  @reserved [1, 2, 3, 4, 5, 6, 7, 8, 11, 12, 13, 19, 20]
 
   # This mapping should be kept in sync with Satellite repo. Message is present
   # in the mapping ONLY if it could be send as an individual message.
   @mapping %{
     SatErrorResp => 0,
-    SatAuthReq => 1,
-    SatAuthResp => 2,
-    SatInStartReplicationReq => 5,
-    SatInStartReplicationResp => 6,
-    SatInStopReplicationReq => 7,
-    SatInStopReplicationResp => 8,
     SatOpLog => 9,
     SatRelation => 10,
-    SatMigrationNotification => 11,
-    SatSubsReq => 12,
-    SatSubsResp => 13,
     SatSubsDataError => 14,
     SatSubsDataBegin => 15,
     SatSubsDataEnd => 16,
     SatShapeDataBegin => 17,
     SatShapeDataEnd => 18,
-    SatUnsubsReq => 19,
-    SatUnsubsResp => 20
+    SatRpcRequest => 21,
+    SatRpcResponse => 22
   }
 
   if Enum.any?(Map.values(@mapping), &(&1 in @reserved)) do
     raise "Cannot use a reserved value as the message tag"
   end
 
+  if elem =
+       Enum.find(Enum.frequencies(Map.values(@mapping)), &match?({_, count} when count > 1, &1)) do
+    {key, count} = elem
+    raise "Cannot have duplicating value tags: #{key} is used #{count} times"
+  end
```

Comment:
```suggestion
  with {tag, messages} <- find_messages_with_duplicate_value_tag(@mapping) do
    raise "Cannot have duplicate value tags: #{tag} is used #{length(messages)} times, namely, by #{inspect messages}"
  end
```

Elsewhere in the module:

```elixir
defp find_messages_with_duplicate_value_tag(map) when is_map(map) do
  map
  |> Enum.group_by(&elem(&1, 1), &elem(&1, 0))
  |> Enum.find(fn {_tag, msgs} -> match?([_, _ | _], msgs) end)
end
```

Reply:
Good idea, but doesn't work since this is module-level compile-time check - we can't use locally defined functions here.

Reply:
This doesn't work, unfortunately, as this is a compile-time check and thus cannot use locally-defined functions, only from other modules, and I don't want to create that yet

Reply:
Oh, I missed that. An anonymous function would work at the module level.